app/src/components/domain/RiskImminentEvents/Pdc/index.tsx

```diff
@@ -45,6 +87,17 @@ function Pdc(props: Props) {
         variant,
     } = props;
 
+    const [
+        activeLayersMapping,
+        setActiveLayersMapping,
+    ] = useState<Record<number, boolean>>(defaultLayersValue);
+
+    const [layers, setLayers] = useState<Record<number, boolean>>(defaultLayersValue);
```

Comment:
Do we need these 2 states? Looks like they are initialized with same value.

Reply:
`activeLayersMapping` persists the state of layers across interactions.
`layers` handles the temporary, interactive behavior of showing or hiding layers.